docs/continuous-delivery/gitops/secret-management/secret-management-gitops.md

```diff
@@ -0,0 +1,189 @@
+- The GitOps repo must contain the name of the Key Vault within the SecretProviderClass
+- Must mount secrets as volumes to allow syncing into Kubernetes Secrets
+- [Missing disconnected scenario support](https://github.com/kubernetes-sigs/secrets-store-csi-driver/issues/446): When the node is offline the SSCSID fails to fetch the secret and thus mounting the volume fails, making scaling and restarting pods not possible while being offline
+- Uses more resources (4 pods; CSI Storage driver and provider) and is a daemonset - not test on RPS / resource usage
+
+### [External Secrets Operator with Azure Key Vault](https://external-secrets.io/)
+
+The External Secrets Operator (ESO) is an open-sourced Kubernetes operator that can read secrets from external secret stores (e.g., Azure Key Vault) and sync those into Kubernetes Secrets. In contrast to the CSI Driver, the ESO controller creates the secrets on the cluster as K8s secrets, instead of mounting them as volumes to pods.
+
+Docs on using ESO Azure Key vault provider [here](https://external-secrets.io/v0.4.4/provider-azure-key-vault/).
+
+Advantages:
+
+- Supports auto rotation of secrets with customizable sync intervals per **secret**.
+- Components are split into different CRDs for namespace (ExternalSecret, SecretStore) and cluster-wide (ClusterSecretStore, ClusterExternalSecret) making syncing more manageable i.r.t. different deployments/pods etc.
+- Service Principal secret for the (Cluster)SecretStores could placed in a namespaced that only the ESO can access (see [Shared ClusterSecretStore](https://external-secrets.io/v0.7.1/guides/multi-tenancy/)).
+- Resource efficient (single pod) - not test on RPS / resource usage.
+- Open source and high contributions, ([GitHub](https://github.com/external-secrets/external-secrets))
+- Mounting Secrets as volumes is supported via K8S's APIs (see [here](https://kubernetes.io/docs/concepts/configuration/secret/#using-secrets-as-files-from-a-pod))
+- Partial disconnected scenario support: As ESO is using native K8s secrets the cluster can be offline, and it does not have any implications towards restarting and scaling pods while being offline
+
+Disadvantages:
+
+- The GitOps repo must contain the name of the Key Vault within the SecretStore / ClusterSecretStore or a ConfigMap linking to it
+- Must create secrets as K8s secrets
+
+## Important Links
+
+- [Sealed Secrets with Flux v2](https://toolkit.fluxcd.io/guides/sealed-secrets/)
+- [Mozilla SOPS with Flux v2](https://toolkit.fluxcd.io/guides/mozilla-sops/)
+- [Secret Management with Argo CD](https://argo-cd.readthedocs.io/en/stable/operator-manual/secret-management/)
+- [Secret management Workflow](https://www.youtube.com/watch?v=-k6HEXaE75k)
+
+## Appendix
+
+### Secrets Rotation on Environment Variables and Mounted Secrets
+
+1. Mapping Secrets via secretKeyRef with environment variables.
```

Comment:
Again, same comment about the numbering, the numbering makes it look like a step by step recipe while these are really just different options